### PR TITLE
simple MPI test for testing autocmake

### PR DESCRIPTION
**Pull request by miroi**
Hereby I am asking experts for the code review of my proposed change - adding the testing of the OpenMPI cmake feature.

### CODE REVIEW

File: test/test.py [hunk 1]
```diff
@@ -72,7 +72,7 @@ def exe(command):
 # ------------------------------------------------------------------------------
 
 
-def configure_build_and_exe(name, setup_command):
+def configure_build_and_exe(name, setup_command, make_command = None, run_command = None):
```
Comment (miroi) [inline test/test.py]: I had to modify this Python function to incorporate user's defined make and run commands.

**Comment by bast** [comment]
Please don't submit PRs which break tests (unless they do so for a good reason).
Every PR consumes a bit of a my time and I don't want to spend time on PRs which
will anyway change few minutes later.

A comment on testing MPI error codes. This is a waste of code lines for most/all MPI installations
where the default is that errors are fatal. The MPI code will just crash and never return
and you will never see the print statements that you have written.

**Comment by miroi** [comment]
I apologize for the inconvenient code. I did change the fortran90 source code accordingly.

**Comment by bast** [comment]
Hi Miro, don't apologize for code :-) This was not criticism. I have checked for the ierrs myself for years
until somebody told me that there is no point.

**Comment by bast** [comment]
Why is the static linking module needed? I mean now. I can imagine that it might be needed later.
Is the `make -j2` really necessary? An API change just for that? This anyway compiles in less than a second.

**Comment by miroi** [comment]
Hi,
Thanks for comments, I simplified affected parts of the code. I had to extend the test.py - configure_build_and_exe - API to incorporate at the 'mpirun -np 2 bin/example' command.

**Comment by bast** [comment]
Thanks Miro. Is the PGI workaround needed?

Also I recommend to change the `run_command` argument and rather introduce
an optional launcher. This launcher could be `mpirun` but later could also
be something else e. g. `valgrind`.

My code review possibly seems strict but this is for a hopefully good reason. I recommend
to imagine that you have to pay 10 EUR for every new line of the code
(every new line of code does cost money even though very indirectly).
Once you implement it in 20 lines, ask yourself: can I do it in 10 lines? Once you have
reduced it in 10 lines, ask yourself: can I do it in 5 lines? Etc. Some problems
end up being solved in 0 lines. The less the better since every line costs maintenance.

**Comment by miroi** [comment]
I don't understand, please how shall I modify the function  "configure_build_and_exe(name, setup_command, run_command=None)" ?

**Comment by bast** [comment]
I would instead of an optional `run_command` arg use an optional `launcher` arg
which - if defined - is placed in front of the executable. The advantages are:
- more general and allows to use other launchers than `mpirun -np 2`
- less and simpler code inside `configure_build_and_exe`

**Comment by miroi** [comment]
Is the code  good ? Can I do rebase ?

**Comment by bast** [comment]
Sure!

**Comment by miroi** [comment]
Ah, after another accepted PR I have to update, fix conflicts and rebase this branch again.

**Comment by bast** [comment]
Thanks!

**Comment by miroi** [comment]
What a good, sweet feeling it is when the PR is accepted after good code review and discussions ! Long live high-quality scientific code development !

**Comment by bast** [comment]
:-)

**Pull request merged**